### ISSUE TITLE
Pipenv installs dependencies twice in test environment (redundant)

### ISSUE THREAD

**Issue by alexandre-paroissien** [heroku/heroku-buildpack-python#824]
Hi it seems that pipenv is running twice in test environment, once for dependencies (pipenv's "packages") and once for test dependencies (pipenv's "packages" + "dev-packages")

https://github.com/heroku/heroku-buildpack-python/blob/master/bin/steps/pipenv

```
-----> Postgresql done

-----> Python app detected

-----> Installing dependencies with Pipenv 2018.5.18…

   Installing dependencies from Pipfile.lock (7877d8)…

-----> Installing test dependencies…

   Installing dependencies from Pipfile.lock (7877d8)…

.-----> Running test-setup command python manage.py migrate --settings=config.heroku_ci_test_settings..
```

The issue happens when running the tests with Heroku CI, the test setup will run "pipenv install" first and then "pipenv install --dev" which is redundant (reinstalls the packages from the first command) and running these two commands is equivalent to running "pipenv install --dev" only

```
   Install all dependencies for a project (including dev):
   $ pipenv install --dev
```

https://github.com/pypa/pipenv

I was able to reproduce this issue with "heroku/python" buildpack, as well as "https://github.com/heroku/heroku-buildpack-python.git" buildpack

But the issue is not when the buildpack is used by the dyno (the dependencies are installed only once during the build which is correct)

How can I only install pipenv's "packages" in test environment?

A quick fix for my issue was to fork the buildpack, comment the unwanted dev-dependencies, and to use it as the test buildpack in app.json

https://github.com/jb-engine/heroku-buildpack-python/commit/22657fd2f9320d4aaf515d46a718df9e59f20d0a

So in my case I wanna run only "pipenv install" for the tests, I am thinking there should be an option to only install main dependencies for those who want it (but by default it should install dev dependencies for everyone else for compatibility with current behaviour, but it should not run both commands because it is redundant and can be quite slow)

**Comment by alexandre-paroissien** [comment]
@CaseyFaist Hello there, I just tested again with "heroku/python" release as well as latest master branch release (ff42f31) 
Unfortunately I am still getting the dependencies installed twice in Heroku Tests as you can see here:

![image](https://user-images.githubusercontent.com/34052139/65035463-7346d880-d973-11e9-9026-0016806d6940.png)

`pipenv install --dev` would give exactly the same result as `pipenv install; pipenv install --dev` except it would save a lot of computing time and bandwidth on the Test Dynos

Were you able to identify this issue? Thanks

**Comment by alexandre-paroissien** [comment]
@CaseyFaist I opened a PR here: #851 
Let me know if I should change or improve anything

**Comment by codingjoe** [comment]
Isn't this solved in #861?

**Comment by CaseyFaist** [comment]
Yep, had to temporarily roll it back with some other changes though which is how this response got lost. Closing now
